.rhdh/scripts/install-registry-proxy-images.sh

```diff
@@ -16,164 +16,185 @@
-  echo "Please install jq 1.2+ from an RPM or https://pypi.org/project/jq/"
-  exit 1
+
+# Get OCP version and architecture
+OCP_VER="v$(oc version -o json | jq -r '.openshiftVersion' | sed -r -e 's#([0-9]+\.[0-9]+)\..+#\1#')"
+OCP_ARCH="$(oc version -o json | jq -r '.serverVersion.platform' | sed -r -e 's#linux/##')"
+[[ $OCP_ARCH == "amd64" ]] && OCP_ARCH="x86_64"
+
+# Default IIB source
+UPSTREAM_IIB="quay.io/rhdh/iib:latest-${OCP_VER}-${OCP_ARCH}"
+
+# Parse arguments
+TO_INSTALL=""
+IMAGES=()
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        '--install-operator') TO_INSTALL="$2"; shift ;;
+        '--next') UPSTREAM_IIB="quay.io/rhdh/iib:next-${OCP_VER}-${OCP_ARCH}" ;;
+        '--latest') UPSTREAM_IIB="quay.io/rhdh/iib:latest-${OCP_VER}-${OCP_ARCH}" ;;
+        '-h'|'--help') usage; exit 0 ;;
+        *) IMAGES+=("$1") ;;
+    esac
+    shift
+done
+
+# If no images were provided, default to IIB
+if [ ${#IMAGES[@]} -eq 0 ]; then
+    IMAGES+=("$UPSTREAM_IIB")
 fi
 
-# Check we're logged into a cluster
-if ! oc whoami > /dev/null 2>&1; then
-  echo "Not logged into an OpenShift cluster"
-  exit 1
+# Enable and get the internal registry route dynamically
+infof "[INFO] Enabling default route for internal registry if not already enabled..."
+oc patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
+
+infof "[INFO] Retrieving internal registry route..."
+INTERNAL_REGISTRY=$(oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
+
+if [[ -z "$INTERNAL_REGISTRY" ]]; then
+    errorf "[ERROR] Unable to retrieve internal registry route."
+    exit 1
 fi
 
-# Create the working directory
-mkdir -p "$WORKDIR"
-
-# Step 1: Pull the IIB image and extract catalog information
-echo "Pulling IIB image and extracting catalog information..."
-skopeo copy --all "docker://$IIB_IMAGE" "oci:$WORKDIR/iib"
-
-# Unpack the OCI image
-cd "$WORKDIR"
-echo "Listing contents of blobs/sha256/ for debugging..."
-ls -R "$WORKDIR/iib/blobs/sha256/"
-
-# Inspect index.json to locate manifests
-indexJson="$WORKDIR/iib/index.json"
-if [[ -f $indexJson ]]; then
-    manifests=$(jq -r '.manifests[].digest' "$indexJson")
-    echo "Found the following manifests: $manifests"
-    
-    images=()
-    
-    for manifest in $manifests; do
-        echo "Checking manifest: $manifest"
-        manifestDir="$WORKDIR/iib/blobs/sha256/${manifest#*:}"
-        
-        manifestJson="$manifestDir/manifest.json"
-        if [[ -f $manifestJson ]]; then
-            layers=$(jq -r '.[].layers[].digest' "$manifestJson")
-            config=$(jq -r '.[].config.digest' "$manifestJson")
-            images+=($layers $config)
+infof "[INFO] Using internal registry: ${INTERNAL_REGISTRY}"
+
+# Authenticate using the external route for image registry
+infof "[INFO] Authenticating to the external registry route..."
+REGISTRY_URL="${INTERNAL_REGISTRY}"  # Remove the https:// part
```

Comment:
comment says to remove the protocol, but you're not doing that. Did you mean...

```suggestion
REGISTRY_URL="${INTERNAL_REGISTRY#*://}"  # Remove the https:// part

```